packages/core/src/tests/root.test.ts

```diff
@@ -0,0 +1,90 @@
+import {createRemoteRoot} from '../root';
+
+describe('root', () => {
+  describe('createComponent', () => {
```

Comment:
Might be nice to have a test that ensures that `createText` is always allowed, even if `components` is defined.